### PR TITLE
Generate Ethereum Private key using BIP32/BIP44

### PR DESCRIPTION
**Pull request by D4nte**

### CODE REVIEW

File: src/ethereum/wallet.rs [hunk 1]
```diff
@@ -16,6 +16,16 @@ use num::BigUint;
 use num256::Uint256;
 use url::Url;
 
+use anyhow::anyhow;
```
Comment (luckysori) [inline src/ethereum/wallet.rs:19]: :upside_down_face: I prefer it if all the `use` statements are adjacent and the `pub use` statements appear in a separate block.

File: src/ethereum/wallet.rs [hunk 2]
```diff
@@ -137,6 +165,7 @@ impl Wallet {
         chain_id: ChainId,
     ) -> anyhow::Result<Hash> {
         self.assert_chain(chain_id).await?;
+        let private_key = self.private_key();
```
Comment (luckysori) [inline src/ethereum/wallet.rs:168]: :question: I don't see why this needed changing.

File: src/ethereum/wallet.rs [hunk 3]
```diff
@@ -70,6 +72,32 @@ impl Wallet {
         }
     }
 
+    pub fn private_key_from_seed(seed: &Seed) -> anyhow::Result<clarity::PrivateKey> {
+        let private_key = Self::root_extended_private_key_from_seed(seed)?
```
Comment (luckysori) [inline src/ethereum/wallet.rs:76]: :question: How do we know what `N` we are using when calling `root_extended_private_key_from_seed`? I am surprised that this compiles :open_mouth:

File: src/ethereum/wallet.rs (hunk 1 above)
Comment (D4nte) [reply src/ethereum/wallet.rs]: me too :)

File: src/ethereum/wallet.rs (hunk 3 above)
Comment (D4nte) [reply src/ethereum/wallet.rs:76]: :) The `N` does not matter to me so I am happy it compiles. I thought the same than you.

File: src/ethereum/wallet.rs (hunk 2 above)
Comment (D4nte) [reply src/ethereum/wallet.rs:168]: It's a function call now.

Comment (D4nte) [reply src/ethereum/wallet.rs]: But actually it's an unnecessary change, now fixed.

**Pull request merged**